pkg/koordlet/metricsadvisor/collectors/coldmemoryresource/cold_page_collector.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package coldmemoryresource
+
+import (
+	"go.uber.org/atomic"
+
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/metricsadvisor/framework"
+	koordletutil "github.com/koordinator-sh/koordinator/pkg/koordlet/util"
+)
+
+const (
+	CollectorName = "coldPageCollector"
+)
+
+type nonCollector struct {
+}
+
+func New(opt *framework.Options) framework.Collector {
+	// check whether support kidled cold page info collector
+	if koordletutil.IsKidledSupported() {
+		koordletutil.IsSupportColdMemory = true
```

Comment:
Be careful about the direct assignment of another pkg's global var.
You can follow the usage of the var `IsUseCgroupV2`, and wrap the assignment inside the `IsColdMemorySupported()`.